src/main/kotlin/yjh/cstar/engine/application/GameEngineService.kt

```diff
@@ -59,25 +63,23 @@ class GameEngineService(
                 logger.info { "[WARN] busy waiting..." }
                 gameAnswerPollRepository.poll(roomId, quiz.id)
                     ?.takeIf {
-                        it.answer.replace(" ", "")
-                            .equals(quiz.answer.replace(" ", ""), ignoreCase = true)
+                        answerValidationService.validateAnswer(it.answer, quiz.answer)
                     }
                     ?.let { result ->
-                        updateScore(ranking, result.playerId)
+                        scoreManagementService.updateScore(ranking, result.playerId)
```

Comment:
scoreManagementService를 @Service 어노테이션을 사용해서 Bean(싱글톤)으로 관리하신 이유가 있을까요??!

현재 점수를 관리하는 서비스에게 "직접" 상태 변경이 가능한 ranking 객체를 넘겨 주고 있습니다.
이렇게 되면 코드 상태 변경의 추적이 어려워지고(디버깅) ranking 이라는 상태 변경 객체를 외부에서 실수로 변경해버릴 수 도 있을 것 같아요!

그래서 차라리 Score 라는 별도의 클래스를 만들고, 필드값으로 score이라는 순위를 기록할 수 있는 상태 필드를 만들어서 사용해보는 것은 어떨까요?!

```suggestion
val score = Score() // 이 부분은 스레드 시작 시점에 선언해야함.
score.update(result.playerId)
```

이렇게 각@Async가 시작될때마다(각 방에서 게임이 시작될 때마다) 각각 Score라는 객체를 생성하고 해당 객체에게 점수를 업데이트 해줘 라는 메세지를 던져주면 실제로 그 내부에서 알아서 점수를 집계해줄 수 있습니다.

이렇게 하면 외부에서 Score 내부에 있는 상태 변경 필드에 실수로 접근하지 않을 수 있고, 실제 내부 동작(TreeMap or Redis 등)에 대해서도 알 수 없게 됩니다!!

Reply:
우선 자세한 답변을 주셔서 감사합니다. 😁 제가 scoreManagementService를 생성한 이유는 추후 여러 점수 관리 방법이 도입될 것을 가정하고 만든 것이었습니다. 무엇보다 현재 scoreManagementService에 상태 변경이 가능한 ranking을 넘겨주고 있는데, 이 방식을 Ranking이라는 도메인 객체를 추가로 생성하여 그 안에서 관련 메서드를 통해 집계하는 쪽으로 수정해보겠습니다.🔥🔥  말씀하신 것처럼 실제 내부 동작은 알 수 없으므로, 추후 다른 기술로 변경될 때에도 유지보수가 용이해질 것 같습니다. 감사합니다!

Reply:
만약 Ranking과 같은 상태 저장을 가지는 클래스를 Infrastructure라고도 볼 수 있을 것 같아요,
그래서 Ranking을 로컬 스토리지라고 생각한다면 추후에는 외부 스토리지로 Redis를 사용하는식으로 개선할 수 있을 것 같습니다!!

Reply:
그렇게 생각해볼 수도 있겠네요! 많이 배웁니다!!